test/unittests/test_agent_tools.py

```
@@ -0,0 +1,247 @@
+    result: int = Field(..., description="Sum of a and b.")
+
+
+def add_logic(args: AddArgs) -> AddOutput:
+    return AddOutput(result=args.a + args.b)
+
+
+def failing_logic(args: AddArgs) -> AddOutput:
+    raise RuntimeError("intentional failure")
+
+
+class MathToolBox(MathToolBox if False else object):
```

Ruff (0.15.6)

[error] Undefined name `MathToolBox`

(F821)